flink-optimizer/src/main/java/org/apache/flink/optimizer/traversals/RangePartitionRewriter.java

```diff
@@ -221,7 +223,7 @@ public void postVisit(PlanNode node) {
 		prRemoverNode.setParallelism(targetParallelism);
 		prPlanNode.setParallelism(targetParallelism);
 		GlobalProperties globalProperties = new GlobalProperties();
-		globalProperties.setRangePartitioned(new Ordering(0, null, Order.ASCENDING));
+		globalProperties.setRangePartitioned(new Ordering(0, null, Order.ASCENDING), channel.getDataDistribution());
```

Comment:
can you pass `null` instead of `channel.getDataDistribution()` to make it clear that no provided data distribution is available?